Lib/test/test_asyncio/test_tasks.py

```diff
@@ -877,10 +893,51 @@ async def inner():
 
             inner_task = self.new_task(loop, inner())
 
-            with self.assertRaises(asyncio.TimeoutError):
-                await asyncio.wait_for(inner_task, timeout=0.1)
+            with self.assertRaises(asyncio.TimeoutError) as cm:
```

Comment:
Can these assertions be moved outside of the coroutine and after the call to `loop.run_until_complete()`? One advantage of doing it that way is that it's easier to see that your assertions will always be exercised, as opposed to skipping them due to e.g. an early return. It also makes the coroutine smaller. This comment holds for the other test cases as well.